src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,387 @@
+    end
+
+  private
+
+    def fix_keys
+      fix_collection_keys(data.data)
+      data.data.each do |entry|
+        if entry[:value].is_a?(AugeasTreeValue)
+          fix_collection_keys(entry[:value].tree.data)
+        end
+      end
+    end
+
+    def fix_collection_keys(entries)
+      entries.each do |entry|
+        if COLLECTION_KEYS.include?(entry[:key])
+          entry[:key] += "[]" unless entry[:key] =~ /.*\[\]$/
+        end
+      end
+    end
+
+    # class to manage ntp entries as a collection.
+    #
+    # Each element of the collection is a subclass of Record.
+    # A Record object is a wrapper of an AugeasElement.
+    class RecordCollection
+      include Enumerable
+
+      def initialize(augeas_tree)
+        @augeas_tree = augeas_tree
+      end
+
+      def each
+        record_entries.each do |augeas_element|
+          yield Record.new_from_augeas(augeas_element)
+        end
+      end
+
+      # Adds a new Record object to the collection.
+      # @param [Record] record
+      def add(record)
+        @augeas_tree.data << record.augeas
+      end
+
+      alias_method :<<, :add
+
+      # Removes a Record object from the collection.
+      # @param [Record] record
+      def delete(record)
+        matcher = Matcher.new do |k, v|
+          k == record.augeas[:key] &&
+            v == record.augeas[:value]
+        end
+        @augeas_tree.delete(matcher)
+      end
+
+      def delete_if(&block)
+        records = select(&block)
+        records.each { |record| delete(record) }
+      end
+
+      def empty?
+        count == 0
+      end
+
+      def ==(other)
+        other.to_a == to_a
+      end
+
+    private
+
+      def record_entries
+        matcher = Matcher.new do |k, _v|
+          RECORD_ENTRIES.include?(k.gsub("[]", ""))
+        end
+        @augeas_tree.select(matcher)
+      end
+    end
+
+    # Base class to represent a general ntp entry.
+    #
+    # This class is an AugeasElement wrapper. A Record
+    # has a value, and could also contains a comment and
+    # options. Its contain is stored in an AugeasElement.
+    #
+    # Each ntp entry type has different interpretation
+    # for its options in the AugeasElement. For each one,
+    # a Record subclass is created.
+    class Record
+      # Creates the corresponding subclass object according
+      # to its AugeasElement key.
+      # @param [String] key
+      def self.new_from_augeas(augeas_entry)
+        record = record_class(augeas_entry[:key]).new
+        record.augeas = augeas_entry
+        record
+      end
+
+      # Returns the corresponding subclass
+      # @param [string] key
+      def self.record_class(key)
+        entry_type = key.gsub("[]", "")
+        record_class = ["::CFA::NtpConf::", entry_type.capitalize, "Record"].join
+        Kernel.const_get(record_class)
+      end
+
+      def initialize(key = nil)
+        @augeas = { key: key, value: nil }
+      end
+
+      attr_accessor :augeas
```

Comment:
is it good idea to have accessor over whole `@augeas` ? I think it is really easy to make class very inconsistent. Any change to `:key` can be very critical and any change to `:value` can kill whole subtree concept introduced by other methods.
What I suggest is to have `key, value and value=` methods.